class-unlist-posts.php

```diff
@@ -149,9 +150,35 @@ public function hide_post_from_searchengines() {
 
 			$hidden_posts = get_option( 'unlist_posts', array() );
 
+			// wp_no_robots is deprecated since WP 5.7.
+			if ( function_exists( 'wp_robots_no_robots' ) ) {
+				return;
+			}
+
+			if ( in_array( get_the_ID(), $hidden_posts, true ) && false !== get_the_ID() ) {
+				wp_no_robots();
+			}
+		}
+
+		/**
+		 * This directive tells web robots not to index the page content if the page is unlisted.
+		 *
+		 * @since  1.1.4
+		 * @param  Array $robots Associative array of robots directives.
+		 */
+		public function hide_post_page_no_robots( $robots ) {
+
+			// Bail if posts unlists is disabled.
+			if ( false === $this->allow_post_unlist() ) {
+				return false;
```

Comment:
Should this be `return $robots;`?

```suggestion
				return $robots;
```

Reply:
Updated